### PR TITLE
Added some validation to handle bad URLs

### PR DESCRIPTION
**Pull request by SecGus**
Added some validation to help deal with bad/wrong input URLs (which was causing segmentation faults & crashes).

### CODE REVIEW

**Comment by GeorginaReeder** [comment]
Thanks for your contribution @SecGus , we appreciate it!

We also have a [Discord server](https://discord.com/invite/projectdiscovery), which you’re more than welcome to join. It's a great place to connect with fellow contributors and stay updated with the latest developments!

File: internal/runner/executer.go [hunk 1]
```diff
@@ -23,21 +23,28 @@ func (r *Runner) ExecuteCrawling() error {
 
 	wg := sizedwaitgroup.New(r.options.Parallelism)
 	for _, input := range inputs {
+		if input == "" {
```
Comment (dogancanbakir) [inline internal/runner/executer.go:26]: This check can be done at https://github.com/projectdiscovery/katana/blob/main/internal/runner/options.go#L89

File: internal/runner/executer.go [hunk 2]
```diff
@@ -23,21 +23,28 @@ func (r *Runner) ExecuteCrawling() error {
...
+			gologger.Warning().Msgf("Skipping empty input")
+			continue
+		}
 		if !r.networkpolicy.Validate(input) {
 			gologger.Info().Msgf("Skipping excluded host %s", input)
 			continue
 		}
 		wg.Add()
 		input = addSchemeIfNotExists(input)
+		if r.crawler == nil {
```
Comment (dogancanbakir) [inline internal/runner/executer.go:36]: We can guard against at the beginning of this function. That way, nothing is done if crawler is nil.

**Review by dogancanbakir — CHANGES_REQUESTED** [review]
Left some comments for you. Thanks for the PR!

**Review by dogancanbakir — APPROVED** [review]

**Review by Mzack9999 — APPROVED** [review]

**Pull request merged**